src/Compilers/CSharp/Portable/Lowering/Instrumentation/Instrumenter.cs

```diff
@@ -256,7 +256,7 @@ public virtual BoundStatement InstrumentPatternSwitchStatement(BoundPatternSwitc
         public virtual BoundStatement InstrumentPatternSwitchWhenClauseConditionalGotoBody(BoundExpression original, BoundStatement ifConditionGotoBody)
         {
             Debug.Assert(!original.WasCompilerGenerated);
-            Debug.Assert(original.Syntax.Parent.Kind() == SyntaxKind.WhenClause);
+            Debug.Assert(original.Syntax.FirstAncestorOrSelf<WhenClauseSyntax>() != null);
```

Comment:
Thanks for the suggestion, but this assertion is as intended. I am not intending to assert that is the only situation that requires skipping. We want to place the instrumentation on the enclosing when clause, and this assertion is that such a thing exists.